smaug/core/operator.h

```diff
@@ -76,14 +97,30 @@ class Operator {
     bool tensorsAllConstructed(const std::vector<TensorBase*>& tensors) const;
     bool validateInputsOutputs() const;
 
+    /** An ordered list of input tensors consumed by this operator.
+     *
+     * Operators may assign semantic meaning to specific input tensors at
+     * specific positions in this array, e.g. index 0 is the input data Tensor
+     * and index 1 is the weight Tensor.
+     */
     std::vector<TensorBase*> inputs;
+
+    /** An ordered list of output tensors produced by this operator.
+     *
+     * Like input tensors, operators can assign semantic meaning to tensors at
+     * specific positions here.
+     */
     std::vector<TensorBase*> outputs;
+
     std::string name;
     OpType opType;
+    /** The BGL Vertex corresponding to this Operator. */
     Vertex vertex;
     Workspace* workspace;
+    /** The number of tensors that this operator is waiting on before it can be
+     * scheduled. */
     int numPendingInputs;
-    // Host memory access types for different data.
+    /** The memory interface over which the inputs are expected to arrive. */
```

Comment:
The memory interface over which the inputs/outputs are transferred.

Reply:
This comment will only show up for the `inputMemType` field in Doxygen. I've copied the doc to the other fields.